server/src/test/java/io/crate/integrationtests/PostgresITest.java

```diff
@@ -905,6 +907,38 @@ public void test_each_non_array_pg_type_entry_can_be_joined_with_pg_proc() throw
         }
     }
 
+    @Test
+    public void test_insert_from_unnest_returning_which_inserts_duplicate_keys() throws Exception {
+        try (var conn = DriverManager.getConnection(url(RW))) {
+            var statement = conn.createStatement();
+            statement.execute("create table tbl (id bigint primary key, val text)");
+            var textGenerator = DataTypeTesting.getDataGenerator(DataTypes.STRING);
+            var insertUnnest = conn.prepareStatement(
+                "insert into tbl (id, val) (select id, val from unnest(?, ?) as t (id, val)) returning _id");
+
+            for (int iteration = 0; iteration < randomIntBetween(2, 7); iteration++) {
```

Comment:
do we need more than 2 iterations? couldn't this be tested by adding a duplicate id inside the `ids` array? already

Reply:
I was thinking of the threshold where jdbc switched to binary streaming to have that covered too - but it's actually already randomized in the `PostgresITest` setup, so I can fix it to 2 iterations here :+1:

Reply:
@seut pushed a fixup. I tried to change it to have duplicate ids in just one request, but then the error doesn't occur. I think it only happens if there is no result rows at all.